File: data/readable_format/cpp/file1732.cpp
#include <iostream>
#include <complex>
#include <algorithm>
#include <vector>

// Define DP (Double Precision) as a type alias for double
using DP = double;
using std::vector;
using std::complex;

// Assuming the et array dimension sizes for the example
const int nbnd = 5; // Number of bands, example value
const int nks = 10; // Number of k-points, example value

// Function prototypes
complex<DP> get_alpha_pv(const vector<vector<DP>>& et);
DP mp_min(const vector<vector<DP>>& et);
DP mp_max(const vector<vector<DP>>& et);

int main() {
    // Example initialization of et with some values
    vector<vector<DP>> et(nbnd, vector<DP>(nks, 0.0));
    for (int ibnd = 0; ibnd < nbnd; ++ibnd) {
        for (int iks = 0; iks < nks; ++iks) {
            et[ibnd][iks] = static_cast<DP>(ibnd + iks); // Example data
        }
    }
    
    auto alpha_pv = get_alpha_pv(et);
    std::cout << "get_alpha_pv result: " << alpha_pv.real() << " + " << alpha_pv.imag() << "i" << std::endl;
    
    return 0;
}

complex<DP> get_alpha_pv(const vector<vector<DP>>& et) {
    DP emin = mp_min(et);
    DP emax = mp_max(et);
    
    DP alpha_pv = 2.0 * (emax - emin);
    alpha_pv = std::max(alpha_pv, 1.0e-2);
    
    return complex<DP>(alpha_pv, 0.0);
}

DP mp_min(const vector<vector<DP>>& et) {
    DP global_min = et[0][0];
    for (const auto& row : et) {
        for (const auto& elem : row) {
            global_min = std::min(global_min, elem);
        }
    }
    return global_min;
}

DP mp_max(const vector<vector<DP>>& et) {
    DP global_max = et[0][0];
    for (const auto& row : et) {
        for (const auto& elem : row) {
            global_max = std::max(global_max, elem);
        }
    }
    return global_max;
}

File: data/readable_format/fortran/file1732.f90
program test_get_alpha_pv
  use, intrinsic :: iso_fortran_env, only: real64
  implicit none
  complex(real64) :: result
  integer, parameter :: dp = real64
  integer, parameter :: nbnd = 5, nks = 10
  real(dp) :: et(nbnd, nks)
  integer :: ibnd, iks

  ! Initialize the et array with some values for demonstration purposes
  do iks = 1, nks
     do ibnd = 1, nbnd
        et(ibnd, iks) = real(ibnd + iks, dp)
     end do
  end do

  result = get_alpha_pv(nbnd, nks, et)
  print *, 'get_alpha_pv result:', result

contains

  function get_alpha_pv(nbnd, nks, et) result(alpha_pv_complex)
    integer, intent(in) :: nbnd, nks
    real(dp), intent(in) :: et(nbnd, nks)
    complex(dp) :: alpha_pv_complex
    real(dp) :: emin, emax, alpha_pv
    integer :: ibnd, iks

    emin = et(1, 1)
    emax = et(1, 1)
    
    do iks = 1, nks
       do ibnd = 1, nbnd
          emin = min(emin, et(ibnd, iks))
          emax = max(emax, et(ibnd, iks))
       end do
    end do

    alpha_pv = 2.0_dp * (emax - emin)
    alpha_pv = max(alpha_pv, 1.0e-2_dp)
    alpha_pv_complex = cmplx(alpha_pv, 0.0_dp, kind=dp)
  end function get_alpha_pv

end program test_get_alpha_pv